test/e2e/operator.go

```diff
@@ -59,14 +68,83 @@ func getOperatorLogs(pod v1.Pod) (string, error) {
 	return logs, nil
 }
 
-func failTestAndPrintLogs(t *testing.T, namespace string, err error) {
-	operatorPod := getOperatorPod(t, namespace)
-	logs, logsErr := getOperatorLogs(*operatorPod)
-	if logsErr != nil {
-		t.Errorf("Couldn't get pod logs: %s", logsErr)
+func printOperatorLogs(t *testing.T, namespace string) {
+	logs, err := getOperatorLogs(namespace)
+	if err != nil {
+		t.Errorf("Couldn't get pod logs: %s", err)
 	} else {
 		t.Logf("Last %d lines of log from operator:\n %s", podLogTailLimit, logs)
 	}
+}
+
+func getKubernetesEvents(namespace string) ([]v1beta1.Event, error) {
+	listOptions := metav1.ListOptions{
+		Limit: kubernetesEventsLimit,
+	}
+
+	events, err := framework.Global.KubeClient.EventsV1beta1().Events(namespace).List(listOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	sort.SliceStable(events.Items, func(i, j int) bool {
+		return events.Items[i].CreationTimestamp.Unix() < events.Items[j].CreationTimestamp.Unix()
+	})
+
+	return events.Items, nil
+}
+
+func printKubernetesEvents(t *testing.T, namespace string) {
+	events, err := getKubernetesEvents(namespace)
+	if err != nil {
+		t.Errorf("Couldn't get kubernetes events: %s", err)
+	} else {
+		t.Logf("Last %d events from kubernetes:\n", kubernetesEventsLimit)
+
+		for _, event := range events {
+			t.Logf("%+v\n\n", event)
+		}
+	}
+}
+
+func getKubernetesPods(namespace string) ([]v1.Pod, error) {
+	pods, err := framework.Global.KubeClient.CoreV1().Pods(namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return pods.Items, nil
```

Comment:
could be simpler:
`framework.Global.KubeClient.CoreV1().Pods(namespace).List(metav1.ListOptions{})`